packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -711,7 +638,9 @@ mixin SchedulerBinding on BindingBase, ServicesBinding {
   bool get framesEnabled => _framesEnabled;
 
   bool _framesEnabled = true;
-  void _setFramesEnabledState(bool enabled) {
+
+  @protected
+  void setFramesEnabledState(bool enabled) {
```

Comment:
we'd probably also want a safer api here, to avoid clashes if different parts of the system try to call it.

e.g. something like returning an object representing a lock that you can release (or taking a future and waiting for it to complete), and suspending frames when there are outstanding locks/futures.